src/rpcmining.cpp

```diff
@@ -61,6 +67,43 @@ UniValue getmininginfo(const UniValue& params, bool fHelp)
         obj.pushKV("kernel-diff-sum",MinerStatus.KernelDiffSum);
     }
 
+    int64_t nMinStakeSplitValue = 0;
+    double dEfficiency = 0;
+    SideStakeAlloc vSideStakeAlloc = {};
+
+    // nMinStakeSplitValue and dEfficiency are out parameters.
+    bool fEnableStakeSplit = GetStakeSplitStatusAndParams(nMinStakeSplitValue, dEfficiency);
+
+    // vSideStakeAlloc is an out parameter.
+    bool fEnableSideStaking = GetSideStakingStatusAndAlloc(vSideStakeAlloc);
+
+    stakesplitting.pushKV("stake-splitting-enabled", fEnableStakeSplit);
+    if (fEnableStakeSplit)
+    {
+        stakesplittingparam.pushKV("min-stake-split-value", nMinStakeSplitValue / COIN);
+        stakesplittingparam.pushKV("efficiency", dEfficiency);
+        stakesplitting.pushKV("stake-splitting-params", stakesplittingparam);
+    }
+    obj.pushKV("stake-splitting", stakesplitting);
+
+    sidestaking.pushKV("side-staking-enabled", fEnableSideStaking);
+    if (fEnableSideStaking)
+    {
+        unsigned int i = 0;
+        for (const auto& alloc : vSideStakeAlloc)
+        {
+            sidestakingalloc.pushKV("id", std::to_string(i));
```

Comment:
I think we can drop the id. It's not going to serve a real purpose.

Reply:
I was thinking it would be helpful if someone is parsing it, but if they are using a native JSON parser, it wouldn't be important. Simpler without the ID.